### ISSUE TITLE
SIGSEGV on googleworkspace_group_member when adding a user to a Cloud Identity group

### ISSUE THREAD

**Issue by danielegozzi** [hashicorp/terraform-provider-googleworkspace#216]
### Terraform Version
 $ terraform -v
 Terraform v1.0.11
 on darwin_amd64
 + provider registry.terraform.io/hashicorp/google v4.1.0
 + provider registry.terraform.io/hashicorp/google-beta v4.1.0
 + provider registry.terraform.io/hashicorp/googleworkspace v0.5.2

### Affected Resource(s)
- googleworkspace_group_member

### Terraform Configuration Files
```hcl
resource "googleworkspace_group_member" "members" {
  group_id = replace(google_cloud_identity_group.cloud_identity_groups_for_project[each.value.name].id, "/^groups//", "")
  email = each.value.group_member

  role = "MEMBER"
  for_each = local.iam_groups_members
}

provider "googleworkspace" {
  customer_id = data.google_organization.org.directory_customer_id
  oauth_scopes = [
    "https://www.googleapis.com/auth/admin.directory.group",
    "https://www.googleapis.com/auth/cloud-identity.groups"
  ]
}
```

### Debug Output
https://gist.github.com/danielegozzi/c19212bd3816c04039b0a5b9b75f7778

### Panic Output
(no panic)

### Expected Behavior
No segmentation fault is ever expected.
I'm trying to add a common Workspaces user from another organization to a Cloud Identity group. I anonymized the log output, but we can pretend that example.com is our primary Workspaces organization while goog.example.com is our Google Cloud Platform organization. We prefer to keep two distinct organizations, but sometimes for alerting I need a regular user (which can receive emails) in a Cloud Identity group.

### Actual Behavior
    provider.terraform-provider-googleworkspace_v0.5.2_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1a94a02]

Segmentation fault. In the Admin console the users appear to have been added to the groups as configured, but having failed the plugin execution the same configuration is retried in subsequent runs, failing again.

### Steps to Reproduce
`terraform apply` but I'm not entirely confident of sharing the whole configuration as the behaviour is likely to depend on our specific Google Workspaces/Cloud Identity setup, so I think setting a variable iam_groups_members as follows should be enough
```hcl
locals {
  iam_groups_members = {
    group_name = {
      name = "group_name@goog.example.com"
      group_member = "somebody@example.com"
    } 
  }
}
```

### Important Factoids
- failure happens when adding a user to a Cloud Identity group
- even if my setup is more complex, involving users from another organization, the problem arises even adding a Cloud Identity user to a Cloud Identity group
- `google_cloud_identity_group_memberships` from the google Terraform provider seems unable to add a Workspaces user to a Cloud Identity group, this is why i was trying with `googleworkspace_group_member` from the googleworkspace provider.

### References
n/a AFAIK

**Comment by megan07** [comment]
Hi @danielegozzi! I'm sorry you're experiencing this issue. I understand your hesitation of providing your configuration, however I am unable to reproduce this. I will provide my configuration below, would you mind letting me know where I might be misunderstanding the issue? I've tried both `googleworkspace_group_member` and `googlce_cloud_identity_group_membership` and they both succeeded. I'm sure I'm missing something, so wanted to clarify. Thanks!

```
resource "google_cloud_identity_group" "cloud_identity_group_basic" {
  display_name         = "group_name"
  initial_group_config = "WITH_INITIAL_OWNER"

  parent = "customers/A01b123xz"

  group_key {
      id = "group_name@example.com"
  }

  labels = {
    "cloudidentity.googleapis.com/groups.discussion_forum" = ""
  }
}

resource "googleworkspace_user" "somebody" {
  primary_email = "somebody@example.com"
  password = "my-password"

  name {
    family_name = "New"
    given_name  = "Somebody"
  }
}

resource "googleworkspace_group_member" "somebody" {
  group_id = trimprefix(google_cloud_identity_group.cloud_identity_group_basic.id, "groups/")

  email = googleworkspace_user.somebody.primary_email
  role  = "MEMBER"
}

# resource "google_cloud_identity_group_membership" "cloud_identity_group_membership_basic" {
#   group    = google_cloud_identity_group.cloud_identity_group_basic.id

#   preferred_member_key {
#     id = googleworkspace_user.somebody.primary_email
#   }

#   roles {
#     name = "MEMBER"
#   }

#   roles {
#     name = "MANAGER"
#   }
# }
```

**Comment by danielegozzi** [comment]
Using your confguration maybe I found a reproduction. If I set up two googleworkspace_group_member for the same user, one with role MEMBER, the other with role OWNER, i get the SIGSEGV.

**Comment by megan07** [comment]
Thanks! I was able to reproduce it now and I see the issue. Thanks!

**Comment by megan07** [comment]
Hi @danielegozzi ! This was due to a [fix](https://github.com/hashicorp/terraform-provider-googleworkspace/pull/211) I put in hoping to import members that were accidentally duplicated, or if the original call failed due to eventual consistency. I failed to consider if the member was added with 2 different roles, terraform won't know what to do. I will revert that code so the error shows up and users can fix their configurations as needed. Sorry about that!
